components/core/src/os/users/admincheck.c

```diff
@@ -0,0 +1,90 @@
+#include <windows.h>
+#include <winnt.h>
+#include <ShlObj.h>
+
+#pragma comment (lib,"shell32.lib")
+
+TOKEN_ELEVATION_TYPE s_elevationType = TokenElevationTypeDefault;
+BOOL                 s_bIsAdmin = FALSE;
+
+// returns 0 if an admin
+// returns 1 if not
+int GetElevationState()
+{
+    if (GetProcessElevation(&s_elevationType, &s_bIsAdmin)) {
+        switch(s_elevationType) {
+            // Default user or UAC is disabled
+            case TokenElevationTypeDefault:  
+                if (IsUserAnAdmin()) {
+                    return 0;
```

Comment:
Yeah not a big deal... @raskchanky Returning early is fine but we have 7 return statements so we're returning early, middle and late ;) Having multiple points of returns leaves us more exposed to leaks over time as code is maintained is the school of thought.